sky/templates/kubernetes-port-forward-proxy-command.yaml.j2

```diff
@@ -0,0 +1,25 @@
+#!/usr/bin/env bash
+set -uo pipefail
+
+# Specifies which config file kubectl should use.
+export KUBECONFIG="{{ kube_config_path }}"
+
+# If the port-forward command is already running on the jump pod, kill it
+pgrep -f "kubectl port-forward svc/{{ ssh_jump_name }} {{ local_port }}:22" > /dev/null
+if [ $? -eq 0 ]; then
+    pkill -f "kubectl port-forward svc/{{ ssh_jump_name }} {{ local_port }}:22"
+fi
+
+# Establishes connection between local port and the ssh jump pod
+kubectl port-forward svc/{{ ssh_jump_name }} {{ local_port }}:22 &
+K8S_PID=$!
+trap "kill -9 $K8S_PID" EXIT
+
+# checks if a conection to local_port of ipaddress is established 
+while ! nc -z {{ ipaddress }} {{ local_port }}; do
+    sleep 0.1
+done
+
+# Estalbishes two directional byte streams to handle stdin/stdout between
+# terminal and the jump pod
+socat - tcp:{{ ipaddress }}:{{ local_port }}
```

Comment:
I was trying this our and it silently failed to ssh for a long time before I realized I don't have socat installed.

Is it possible to check if `socat` is installed at the start of the script, raise an error if its not and propagate this error cleanly up to the user? Otherwise, we may want to add a check if `socat` is installed elsewhere in our code...

Reply:
I added a check for `socat` installation and displays error msg and exit at the beginning of the script if it's not installed so that it shows the msg when users attempts to `ssh <k8s-instance-name>` without `socat` installed.

But it doesn't seem like there's a clean way to handle this exit and raise an error msg for every possible ssh session runs within skypilot. So I added another check for `socat` installation in `authentication.py/setup_kubernetes_authentication.py` when 'port-forward' mode is being setup.